### PR TITLE
Do not panic in nexus test poller helper on not found errors

### PR DESCRIPTION
**Pull request by bergundy**
Found here: https://github.com/temporalio/temporal/actions/runs/11473702877/job/31928413916#step:8:8773

### CODE REVIEW

**Comment by bergundy** [comment]
Trying to work around lint-actions failure, looks like there's no way to ignore this specific false negative: https://github.com/rhysd/actionlint/issues/237.

Seems like there's a bug in actionlint that's not properly detecting that this flag is supported in `actions/upload-artifact@v4`. I'll fix the linter somehow in a followup PR.

File: tests/nexus_test_base.go [hunk 1]
```diff
@@ -88,8 +89,9 @@ func (s *NexusTestBaseSuite) versionedNexusTaskPoller(ctx context.Context, taskQ
 			TaskToken: res.TaskToken,
 			Error:     handlerError,
 		})
-		// There's no clean way to propagate this error back to the test that's worthwhile. Panic is good enough.
-		if err != nil && ctx.Err() == nil {
+		// Ignore if context is already cancelled or if the task is not found.
+		if err != nil && ctx.Err() == nil && !errors.As(err, new(*serviceerror.NotFound)) {
```
Comment (rodrigozhou) [inline tests/nexus_test_base.go:93]: If the context was canceled, then `ctx.Err()` returns an "context canceled" error.

Comment (bergundy) [reply tests/nexus_test_base.go:93]: Don't understand the comment. We want to panic only if the context wasn't canceled.

Comment (rodrigozhou) [reply tests/nexus_test_base.go:93]: Oh, right! I misread the code. Makes sense now.

**Review by rodrigozhou — APPROVED** [review]

**Pull request merged**